### DISCUSSION TITLE
GPT Training Memory Estimation - NeMo Practice | Jianbin Chang

### DISCUSSION THREAD

**Discussion by giscus** [alshedivat/al-folio#2577] [Comments]
# GPT Training Memory Estimation - NeMo Practice | Jianbin Chang

Nice blog. 

https://shjwudp.github.io/blog/2023/gpt-training-memory-estimation-nemo-training-practice/

**Comment by AlexBodner** [comment]
Hi Jianbin, awesome article, haven't found one that is more detailed! Do you mind to share the code for the memory estimation? Because I implemented it and I'm getting different results so I'm probably missing something. 
Best wishes!
Alex

**Comment by sowmen** [comment]
What do tensor parallel size (t) and data parallel size (d) mean? For 1 node with 8xA6000 GPUs what would be the values?

**Comment by liviclee** [comment]
Good job, mark.

**Comment by butyuhao** [comment]
Thank you for your blog. You mentioned that ‘The linear projection stores its input activations with size 2sbh’. Why is there a need to multiply by 2 here?
